src/Client.php

```diff
@@ -34,29 +36,42 @@ class Client
 
     private $builder;
 
+    private $signatureEnabled = false;
+
     public function __construct(Builder $builder = null, string $mode = self::MODE_TEST)
     {
         $this->builder = $builder ?: new Builder();
 
         $uri = $mode === self::MODE_TEST ? self::ENDPOINT_TEST : self::ENDPOINT_LIVE;
 
-        $builder->addPlugin(new ClearhausExceptionThrower());
-        $builder->addPlugin(new Plugin\AddHostPlugin(
+        $this->builder->addPlugin(new ErrorPlugin());
+        $this->builder->addPlugin(new Plugin\AddHostPlugin(
             UriFactoryDiscovery::find()->createUri($uri)
         ));
     }
 
+    public function enableSignature()
+    {
+        $this->signatureEnabled = true;
+    }
+
     public function getHttpClient() : HttpMethodsClient
     {
         return $this->builder->build();
     }
 
-    public function setApiKey(string $apiKey)
+    public function useApiKey(string $apiKey)
     {
-        $this->builder->removePlugin(Authentication::class);
+        $this->builder->removePlugin(AuthenticationPlugin::class);
+        $this->builder->removePlugin(SignaturePlugin::class);
 
-        $authenticationPlugin = new Authentication(new BasicAuth($apiKey, ''));
+        $authenticationPlugin = new AuthenticationPlugin(new BasicAuth($apiKey, ''));
         $this->builder->addPlugin($authenticationPlugin);
+
+        if ($this->signatureEnabled) {
```

Comment:
Should this be part of another method? `useApiKey` sound like it should only set the api key (and therefore the `AuthenticationPlugin`)